### PR TITLE
[FIX] sale_stock: display lots on invoices

### PR DESCRIPTION
**Pull request by adwid**
When printing an invoice with tracked products, if there are several
invoices linked to the sale order, the displayed lots may be incorrect

To reproduce the issue:
(Need sale_management)
1. In Settings, enable "Display Lots & Serial Numbers on Invoices"
2. Create two products P01, P02:
    - Storable
    - Tracked by lot
3. Update their quantity (>1)
4. Create and confirm a SO with 1 x P01 and 1 x P02
5. Deliver the products one by one
6. From the SO, create an invoice INV01
7. On INV01, remove the line associated to P02
8. Post INV01
9. From the SO, create and post the second invoice INV02
10. Print one of the invoices

Error: [1] When printing INV01 (which only contains P01), the lot of P02
is also displayed. [2] When printing INV02 (which only contains P02),
the lot of PO02 is not displayed

When rendering an invoice, `_get_invoiced_lot_values` is called and
provides all data about invoiced lots. However, there are two issues:
- [1] it considers all related SML, even the ones related to a product
not present in the invoice
- [2] when computing the date of the last invoice, it uses the same date
for all products in the invoice although it could be different depending
on the product.

OPW-2637107

### CODE REVIEW

**Comment by adwid** [comment]
Hi @smetl, could you review this ?

**Review by wtaferner — COMMENTED** [review]
The retrieval in the start of the whole logic triggered me for the rest I guess it is probably helpful to review and improve the black magic which is working obviously not for all use cases 😄

File: addons/sale_stock/models/account_move.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def _get_invoiced_lot_values(self):
             return []
 
         sale_orders = self.mapped('invoice_line_ids.sale_line_ids.order_id')
-        stock_move_lines = sale_orders.mapped('picking_ids.move_lines.move_line_ids')
+        stock_move_lines = sale_orders.mapped('picking_ids.move_lines.move_line_ids').filtered(lambda sml: sml.product_id in self.invoice_line_ids.product_id)
```
[the remark this answers is not in the document]
Comment (adwid) [reply addons/sale_stock/models/account_move.py]: Yep, I agree
Thx

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**